### PR TITLE
Replaced [u8;32] with account id

### LINKED ISSUE #35: revert untyping AccountId to slice
https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/1fd9e7b03a7b75d0adecfa8cc158d36cc556ef19/polkadex-primitives/src/lib.rs#L12

@haerdib @ksr30 

Please use `AccountId32`, not `[u8; 32]`. I see trouble down the road and I think strong typing is a good thing, we shouldn't allow such workarounds.

If you need to insert it into a hash table, you just add `.encode()` if the main type doesn't implement the necessary trait.

### PR DESCRIPTION
**Pull request by jaxter03**

### CODE REVIEW

File: enclave/src/polkadex.rs [hunk 1]
```diff
@@ -115,32 +118,33 @@ impl PolkadexAccountsStorage {
             accounts: HashMap::new(),
         };
         for account in accounts {
-            in_memory_map
-                .accounts
-                .insert(account.account.current, account.account.proxies);
+            in_memory_map.accounts.insert(
+                account.account.current.using_encoded(blake2_256),
```
Comment (Gauthamastro) [inline enclave/src/polkadex.rs]: Do we really need blake2_256? why don't we simply .encode()?

File: enclave/src/polkadex.rs [hunk 2]
```diff
@@ -115,32 +118,33 @@ impl PolkadexAccountsStorage {
...
+                account.account.proxies,
+            );
         }
         in_memory_map
     }
 
     pub fn add_main_account(&mut self, acc: AccountId) {
-        if self.accounts.contains_key(&acc) {
+        if self.accounts.contains_key(&acc.using_encoded(blake2_256)) {
```
Comment (Gauthamastro) [inline enclave/src/polkadex.rs]: Same as above

File: polkadex-primitives/src/lib.rs [hunk 3]
```diff
@@ -6,22 +6,21 @@ use sgx_tstd as std;
 
 use std::vec::Vec;
 
-use codec::{Encode, Decode};
+use codec::{Decode, Encode};
 use sp_core::crypto::AccountId32;
 
-pub type AccountId = [u8;32];
+pub type AccountId = AccountId32;
```
Comment (Gauthamastro) [inline polkadex-primitives/src/lib.rs]: This is not AccountId32 but AccountId imported from polkadex-primitives of Polkadex Node

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

**Comment by Gauthamastro** [comment]
@ksr30 I think you are not importing the polkadex-primitives of Polkadex Node

File: polkadex-primitives/src/lib.rs (hunk 3 above)
Comment (jaxter03) [reply polkadex-primitives/src/lib.rs]: Yes @Gauthamastro, as soon as that PR is merged ... I will replace it

File: enclave/src/polkadex.rs (hunk 1 above)
Comment (Gauthamastro) [reply enclave/src/polkadex.rs:122]: @ksr30

File: enclave/src/polkadex.rs (hunk 2 above)
Comment (Gauthamastro) [reply enclave/src/polkadex.rs:130]: @ksr30

File: enclave/src/test_proxy.rs [hunk 4]
```diff
@@ -11,24 +13,31 @@ use std::{
...
 pub fn get_dummy_map(storage: &mut SgxMutexGuard<PolkadexAccountsStorage>) {
-    let main_account_one: AccountId = Vec::from("first_account").using_encoded(blake2_256);
-    let main_account_two: AccountId = Vec::from("second_account").using_encoded(blake2_256);
-    let main_account_three: AccountId = Vec::from("third_account").using_encoded(blake2_256);
-    let dummy_account_one: AccountId = Vec::from("first_dummy_account").using_encoded(blake2_256);
-    let dummy_account_two: AccountId = Vec::from("second_dummy_account").using_encoded(blake2_256);
-    let dummy_account_three: AccountId = Vec::from("third_dummy_account").using_encoded(blake2_256);
+    let main_account_one: AccountId =
+        AccountId::new(Vec::from("first_account").using_encoded(blake2_256));
+    let main_account_two: AccountId =
```
Comment (Gauthamastro) [inline enclave/src/test_proxy.rs]: We don't need blake2_256, use .encode()

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**